cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```
@@ -126,6 +127,39 @@ public void deleteRepository(FullHttpRequest request, HttpResponder responder,
                                                               namespaceId));
   }
 
+  /**
+   * Lists applications found in linked repository for the given namespace.
+   * Returns the application name and git-filehash for each application config file found
+   * <pre>
+   * {@code
+   * {
+   *  apps:[
+   *    {
+   *      "name": "xyz",
+   *      "fileHash": "abc"
+   *    }
+   *  ]
+   * }
+   * }
+   *
+   * </pre>
+   */
+  @GET
+  @Path("/apps")
+  public void listApplications(FullHttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId) throws Exception {
+    checkSourceControlFeatureFlag();
+    NamespaceId namespace = validateNamespaceId(namespaceId);
+    try {
+      ListAppsResponse listResponse = sourceControlService.listApps(namespace);
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(listResponse));
+    } catch (AuthenticationConfigException e) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, e.getMessage());
+    } catch (SourceControlFailureException e) {
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage());
```

instead of this, you can just let it propagate up and it will result in the same behavior, 500 response with the exception message as the body.